be/src/exec/stream/state/imt_state_table.cpp

```diff
@@ -0,0 +1,134 @@
+            .version = _imt.olap_table.schema.version,
+            .partition_param = _imt.olap_table.partition,
+            .location_param = _imt.olap_table.location,
+            .nodes_info = _imt.olap_table.nodes_info,
+    };
+}
+
+stream_load::OlapTableSinkParams IMTStateTable::_convert_to_sink_params() {
+    return stream_load::OlapTableSinkParams{
+            .load_id = _imt.load_id,
+
+            .txn_id = _imt.txn_id,
+            .load_channel_timeout_s = 0,
+            .num_replicas = 1,
+            .tuple_id = _imt.olap_table.schema.tuple_desc.id,
+
+            .keys_type = TKeysType::PRIMARY_KEYS,
+            .schema = _imt.olap_table.schema,
+            .partition = _imt.olap_table.partition,
+            .location = _imt.olap_table.location,
+            .nodes_info = _imt.olap_table.nodes_info,
+
+            .write_quorum_type = TWriteQuorumType::MAJORITY,
+            .merge_condition = "",
+            .txn_trace_parent = "",
+
+            .is_lake_table = false,
+            .need_gen_rollup = false,
+            .enable_replicated_storage = false,
+    };
+}
+
+Status IMTStateTable::seek(const Columns& keys, StateTableResult& state_result) const {
+    return seek(keys, _non_keys_field_names, state_result);
+}
+
+Status IMTStateTable::seek(const Columns& keys, const std::vector<uint8_t>& selection,
+                           StateTableResult& state_result) const {
+    return _table_reader->multi_get(keys, selection, _non_keys_field_names, state_result.found,
+                                    &state_result.result_chunk);
+}
+
+Status IMTStateTable::seek(const Columns& keys, const std::vector<std::string>& projection_columns,
+                           StateTableResult& state_result) const {
+    return _table_reader->multi_get(keys, projection_columns, state_result.found, &state_result.result_chunk);
+}
+
+ChunkIteratorPtrOr IMTStateTable::prefix_scan(const Columns& keys, size_t row_idx) const {
+    return prefix_scan(_non_keys_field_names, keys, row_idx);
+}
+
+ChunkIteratorPtrOr IMTStateTable::prefix_scan(const std::vector<std::string>& projection_columns, const Columns& keys,
+                                              size_t row_idx) const {
+    std::vector<const ColumnPredicate*> predicates;
+    predicates.reserve(keys.size());
+    auto datum_tuple = _convert_to_datum_tuple(keys, row_idx);
+    ObjectPool obj_pool;
+    _table_reader->build_eq_predicates(datum_tuple, &predicates, obj_pool);
+    return _table_reader->scan(projection_columns, predicates);
+}
+
+DatumTuple IMTStateTable::_convert_to_datum_tuple(const Columns& keys, size_t row_idx) const {
+    DatumTuple res;
+    res.reserve(keys.size());
+    for (const auto& column : keys) {
+        res.append(column->get(row_idx));
+    }
+    return res;
+}
+
+Status IMTStateTable::write(RuntimeState* state, StreamChunk* chunk) {
+    DCHECK(chunk);
+    DCHECK(_olap_table_sink);
+    if (StreamChunkConverter::has_ops_column(chunk)) {
```

Comment:
What's the difference? Why not strip the ops columns and convert it to a normal chunk ?

Reply:
yet. I add a todo later. Maybe we should add some converts to do this.